### ISSUE TITLE
Pipeline / Module repository

### ISSUE THREAD

**Issue by ewels** [ewels/clusterflow#71]
The code base and user base of Cluster Flow is expanding. Bundling everything with the core package is bulky, and makes it difficult for people to customise things without deviating from the main installation.

Sure, you can use personal directories with custom modules and pipelines, but a) people don't really know that exists and b) it doesn't solve all of the above.

Potential setup:

* Create some kind of repository http://clusterflow.io
* Write code for CF which allows command line querying of what is available
* CF code to download modules and pipelines (choose location?)
* Think about how to implement version control
 * Different repository / repositories?

**Comment by s-andrews** [comment]
Whilst you're listing ideas I'd say it would also be useful to be able to create local repositories per site and let people import modules/pipelines from there to their own workspace.  This would then mean you could lose the global/local distinction as all pipelines would be local ones.

**Comment by ewels** [comment]
Yes, that would be nice! That was kind of my intention with the current setup of the central installation vs. the user directory (which takes preference). But I get the impression that pretty much everyone just uses the centrally installed stuff - if it wasn't possible to run modules and pipelines from the central installation then I guess that would sort that out. A local repository would be less bloated too.

Ok - so what would a local repository look like? Just a directory containing modules and pipelines?

**Comment by s-andrews** [comment]
I'd say the local repository could look exactly as it does now.  All you'd do is to make the central repository be just a repository from which you can fetch modules or pipelines and not something you can run from directly.

**Comment by ewels** [comment]
ok cool. I'll update the top comment with a to-do list.
